### ISSUE TITLE
Web app crashed when opened in private tab

### ISSUE THREAD

**Issue by julianfoad** [binwiederhier/ntfy#208]
* I had the web-app of my self-hosted ntfy instance open in a normal Firefox tab. It was working fine, showing several subscriptions.
* I opened a Firefox private browsing tab and opened the web-app of my self-hosted ntfy instance, <https://ntfy.example.org>.
* The app opened up showing an error page instead of the expected content: "Oh no, ntfy crashed" with this stack trace:
```
DatabaseClosedError: InvalidStateError A mutation operation was attempted on a database that did not allow mutations.
 InvalidStateError: A mutation operation was attempted on a database that did not allow mutations.
  at undefined (DatabaseClosedError: InvalidStateError A mutation operation was attempted on a database that did not allow mutations.:undefined:undefined)
  at undefined ( InvalidStateError: A mutation operation was attempted on a database that did not allow mutations.:undefined:undefined)
  at from (../node_modules/dexie/dist/modern/dexie.min.mjs:16613:1)
  at from (../node_modules/dexie/dist/modern/dexie.min.mjs:16801:1)
  at from (../node_modules/dexie/dist/modern/dexie.min.mjs:15440:1)
  at from (../node_modules/dexie/dist/modern/dexie.min.mjs:10172:1)
  at from (../node_modules/dexie/dist/modern/dexie.min.mjs:10929:1)
  at from (../node_modules/dexie/dist/modern/dexie.min.mjs:11290:1)

DatabaseClosedError: InvalidStateError A mutation operation was attempted on a database that did not allow mutations.
 InvalidStateError: A mutation operation was attempted on a database that did not allow mutations.
  at Ym@https://ntfy.foad.me.uk/static/js/main.3e6ef68a.js:2:950278
  at Nf@https://ntfy.foad.me.uk/static/js/main.3e6ef68a.js:2:883982
  at n@https://ntfy.foad.me.uk/static/js/main.3e6ef68a.js:2:923169
  at P@https://ntfy.foad.me.uk/static/js/main.3e6ef68a.js:2:355181
  at S@https://ntfy.foad.me.uk/static/js/main.3e6ef68a.js:2:354906
  at C@https://ntfy.foad.me.uk/static/js/main.3e6ef68a.js:2:355291
  at Ff@https://ntfy.foad.me.uk/static/js/main.3e6ef68a.js:2:883242
  at im@https://ntfy.foad.me.uk/static/js/main.3e6ef68a.js:2:919841
  at Xm
```
* I can reproduce it at the moment: any time I open it in a private-mode tab it does the same.
* Version: as served by the Docker image id `sha256:2fd60d0cd40195ca2d41e32131f79cdfabc6e3f766bf6acc4efcb5524771ce39`.
* Config: the only config option I set is `--behind-proxy`.
* I am not waiting on a fix: I don't need this to work, I just stumbled across it.

**Comment by binwiederhier** [comment]
I'm closing this as "won't fix" because as I said:

> It would be tremendeous effort to make it work, with very little benefit. Why would anyone use an app with local-only storage in private mode? The use cases are pretty much non-existant.

Hope you understand.

**Comment by julianfoad** [comment]
As I said originally, I have no need for the app to work in private browsing mode: I simply happened to be in that mode and it crashed. When it crashed there was no clue that this was the reason and a known limitation. Then (being the curious type) I investigated further.

This seems ideal:

> I could make it crash nicer showing why/that it doesn't work...

But now that I myself know how to avoid the crash, it's nowhere on my priority wish list. I won't argue any further.

**Comment by binwiederhier** [comment]
Re-opening as this _just_ happened to another user. So I suppose I'll add a nicer error message :-D

**Comment by binwiederhier** [comment]
Did this. I hate it, but I also don't want to re-implement everything for a use case that doesn't exist. 

![220429 20-47-47 Selection 001](https://user-images.githubusercontent.com/664597/166084102-fdc34136-0e62-4892-9545-9a2c0b2d9cce.png)

**Comment by julianfoad** [comment]
Compared with a crash that's night-and-day better. Thanks!
